### ISSUE TITLE
[odc-stac] Create one function to load STAC into xarrays

### ISSUE THREAD

**Issue by gadomski** [opendatacube/datacube-core#1192]
Currently it takes two steps to go from STAC items to an xarray. It'd be nice to have one top-level function that ties together `odc.stac.stac2ds` and `odc.stac.dc_load`.

### Expected behavior

```python
from odc import stac
items = load_some_stac_items()
xarray_data = stac.load(items, **config)
```

### Actual behavior

```python
from odc import stac
items = load_some_stac_items()
datasets = stac.stac2dc(items, **config_one)
xarray_data = stac.dc_load(datasets, **config_two)
```

**Comment by Kirill888** [comment]
For planetary computer specifically there is another step in the *actual behavior* - signing of urls within a `Dataset` object. Several options to handle that:

1. Sign earlier at STAC.Item level (I had some issues with that, signed Items had assets that did not point back to Item and so extensions broke)
2. Configure signing inside `stac.load`
3. Rewrite urls into `/vsiaz/` when appropriate and let GDAL deal with credentials

**Comment by Kirill888** [comment]
I have done some exploration regarding the use of `/vsiaz/` urls to let GDAL deal with signing. This works but only after this fix in GDAL;

https://github.com/OSGeo/gdal/commit/6888f976933c444a7a12d7d65bfbeeaaa696d7b5

So it works in `3.3.X` series, but not in `3.2.X` series of GDAL. Version of GDAL currently installed on planetary computer is `3.2.1`, and without that fix range requests are broken as they do not have SAS token appended so azure returns 404.

**Comment by gadomski** [comment]
> Sign earlier at STAC.Item level (I had some issues with that, signed Items had assets that did not point back to Item and so extensions broke)

Can you provide an example of this breakage? It _seems_ like pre-signing the STAC items should be good enough and would make this path pretty straightforward.

**Comment by Kirill888** [comment]
@gadomski actually it's my code that broke, as I assumed owner was always set, extensions did work fine on "detached" assets.

and if you mean example of how signing broke, I made this issue:

https://github.com/microsoft/planetary-computer-sdk-for-python/issues/25
